src/universal-app/prerender.ts

```diff
@@ -16,10 +16,12 @@ const indexHtmlPath = require.resolve('./index.html');
 const result = renderModuleFactory(
     KitchenSinkRootServerModuleNgFactory,
     {document: readFileSync(indexHtmlPath, 'utf-8')});
+const runfilesDir = process.env.RUNFILES_DIR as string;
+const bazelOutDir = dirname(runfilesDir.replace(/sandbox\/[a-z0-9]+-sandbox\/[0-9]+\//, ''));
```

Comment:
I think it would be better to write to `TEST_UNDECLARED_OUTPUTS_DIR` instead of doing some sandbox magic here.

https://docs.bazel.build/versions/main/test-encyclopedia.html

Reply:
i got the point, i don't like the "sandbox magic" either
i was thinking that a better solution could be to run the test using `bazel run` rather than `bazel test` to write to bin dir
i will fix this and update the DEBUG guide accordingly

Reply:
👍